extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/RuleEliminateMarker.scala

```diff
@@ -17,11 +17,34 @@
 
 package org.apache.kyuubi.plugin.spark.authz.util
 
-import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.catalog.HiveTableRelation
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, Statistics}
 import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.command.DDLUtils
+import org.apache.spark.sql.internal.SQLConf
 
 class RuleEliminateMarker extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = {
-    plan.transformUp { case rf: RowFilterAndDataMaskingMarker => rf.table }
+    plan.transformUp {
+      case rf: RowFilterAndDataMaskingMarker =>
+        val table = rf.table
+        table match {
+          case relation: HiveTableRelation =>
```

Comment:
see `hasResolvedHiveTable/getHiveTable`

Reply:
updated

Reply:
> TODO: remove this after we completely make hive as a data source.

FYI, that's the reason why I suggest using these methods, please update other places accordingly?